scripts/Run-Self-Hosted-Runner-Test.ps1

```diff
@@ -0,0 +1,261 @@
+    $UserModeDumpFilePath = Join-Path $UserModeDumpFolder $UserModeDumpFileName
+
+    if ($VerbosePreference -eq 'Continue') {
+        Write-Log "User mode Dumpfile name: $UserModeDumpFileName"
+        Write-Log "User mode Dumpfile Path: $UserModeDumpFilePath"
+    }
+
+    # Get the available free space at this point in case the test creates files of it's own.
+    # (useful in investigating user and/or kernel dump file creation failures).
+    $DriveFreeSpaceGB = GetDriveFreeSpaceGB -DriveSpecification $Env:SystemDrive
+    Write-Log "Current available disk space: $DriveFreeSpaceGB GB`n"
+
+    Write-Log "Creating User mode dump @ $UserModeDumpFilePath"
+    $ProcDumpArguments = "-r -ma $($TestProcess.Id) $UserModeDumpFilePath"
+    Write-Log "Dump Command: $ProcDumpBinaryPath $ProcDumpArguments"
+    $ProcDumpProcess = Start-Process -NoNewWindow `
+        -FilePath $ProcDumpBinaryPath `
+        -ArgumentList $ProcDumpArguments `
+        -Wait -PassThru
+    Write-Log "Waiting for user mode dump to complete..."
+    $ProcDumpProcess.WaitForExit()
+
+    # Get procdump64.exe's exit code.
+    $ExitCode = $($ProcDumpProcess.ExitCode)
+    Write-Log "$ProcDumpBinaryPath completed with exit code: $ExitCode`n"
+
+    # Flush disk buffers to ensure user mode dump data is completely written to disk.
+    Write-Log "User mode dump completed. Flushing disk buffers..."
+    Write-VolumeCache -DriveLetter C
+
+    # Wait for a bit for things to settle down to ensure the user mode dump is completely flushed.
+    Start-Sleep -seconds 10
+
+    # Paranoia check: Make sure a (seemingly) valid dump file _is_ created.
+    $UserModeDumpSizeMB =
+        (((Get-ItemProperty -Path $UserModeDumpFilePath).Length) /1MB).ToString("F2")
+    if ($UserModeDumpSizeMB -eq 0) {
+        Write-Log "* WARNING * User mode dump $UserModeDumpFilePath NOT CREATED"
+    } else {
+        Write-Log "`n`n"
+        Write-Log "Created $UserModeDumpFilePath, size: $UserModeDumpSizeMB MB"
+        Write-Log "`n`n"
+    }
+
+    if ($NeedKernelDump) {
+        Write-Log "Creating kernel dump...`n"
+        # Wait a bit for the above message to show up in the log.
+        Start-Sleep -seconds 5
+
+        # This will/should not return (test system will/should bluescreen and reboot).
+        Start-Process -NoNewWindow -Wait -FilePath $NotMyFaultBinaryPath -ArgumentList "/crash"
+
+        # If we get here, notmyfault64.exe failed for some reason. Kill the hung process, throw error.
+        $TestProcess.Kill()
+        ThrowWithErrorMessage `
+            -ErrorMessage "*** ERROR *** $($PSCommandPath): kernel mode dump creation FAILED"
+    } else {
+        Write-Log "`n`n"
+        Write-Log "Kernel dump not needed, killing test process $($TestProcess.ProcessName)..."
+        $TestProcess.Kill()
+        Write-Log "`n`n"
+
+        $ErrorMessage = "*** ERROR *** $($PSCommandPath): $FullTestCommandSpec.exe exceeded test hang timout of " +
```

Comment:
this message seems a duplicate of line 169.
Can you include the sample new output in the PR description?

Reply:
We get here only if the test timeout expires _AND_ the kernel dump is disabled (a very rare case, IMO).  We still need to `throw` with a suitable error message.  The slight duplication here makes sense in this very specific (and extremely rare) scenario.